Track kills and deaths, show in score screen at end of match

nathanleiby opened the pull request:
![score-screen-withs-kills-deaths2](https://user-images.githubusercontent.com/102242/80769306-a18fb780-8b01-11ea-8cce-fe3f875934ff.gif)

nathanleiby committed:
track kills and deaths

nathanleiby committed:
link CombatManager to child prefabs

nathanleiby committed:
gitignore BuildVersions folder

nathanleiby committed:
show player scores at end of combat

peternga commented on Assets/Scripts/EndCombatUI.cs:
```diff
@@ -1,16 +1,50 @@
+
     public GameObject panel;
 
+    const int MAX_PLAYERS = 4;
```
Is there a place we can centralize this data? Seems fragile to have to redefine it in every script

nathanleiby replied:
Moved it to a singleton. Didn't look into this too deeply, just copied an approach from the Unity wiki. https://github.com/CleverGameDev/Cleverfall/commit/42ec046302fc3fcd8ba555c4258caf6733c5d1d2

peternga commented on Assets/Scripts/EndCombatUI.cs:
```diff
@@ -1,16 +1,50 @@
+                int kills = humanPlayers[i].GetKills();
+                int deaths = humanPlayers[i].GetDeaths();
+                if ((kills - deaths) > bestScore) {
+                    winningPlayerIndex = humanPlayers[i].playerInput.playerIndex;
```
Don't we also need to update bestScore?

nathanleiby replied:
derp, fixed here 42ec046302fc3fcd8ba555c4258caf6733c5d1d2

peternga commented on Assets/Scripts/EndCombatUI.cs:
```diff
@@ -1,16 +1,50 @@
+        int winningPlayerIndex = -1;
+
+        for (int i = 0; i < humanPlayers.Length; i++) {
+            if (humanPlayers[i] == null) {
```
What happens if we just skip the players that are null?

nathanleiby replied:
Currently, it would show scores for them (0 kills 0 deaths). As we make the score screen fancier, it will make even less sense b/c they won't have a selected character image to show, etc.

nathanleiby committed:
pr feedback

- add singleton for global vars
	- Unity community suggests using Singleton instead of static
	https://answers.unity.com/questions/50716/how-to-make-a-global-variable-in-unity.html
	- Singleton base code from: http://wiki.unity3d.com/index.php/Singleton
- actually update best Score
